docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,85 @@
+# Enchantments
+
+Enchantments are enchancements that can be applied to tools and other items. As of 1.21, enchantments are stored on items as [Data Components], are defined in JSON, and are comprised of Enchantment Components.
+
+A new enchantment can be added by creating a JSON file in your namespace's `enchantment` datapack subfolder. For example, to create an enchantment called `examplemod:example_enchant`, one could create the file `data/examplemod/enchantment/example_enchantment.json`. Enchantments added this way are automatically registered into the game as part of the data pack loading process.
+
+The JSON format for Enchantments is summarized on the [Enchantment definition Minecraft wiki page].
+
+## Enchantment Components
+Enchantment Components are specially-registered [Data Components] that determine what an enchantment does. Enchantment Component Types must be [registered] to `BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE`.
+:::note
+There are many already existing Enchantment Component Types in vanilla that can be used to create enchantment logic.
+:::
+
+For simple enchantments that act as a marker for custom logic, `DataComponentType<Unit>` may be sufficient. Such a component type could be registered, for example, using a DeferredRegister:
+```java
+// In some registration class
+public static final DeferredRegister<DataComponentType<?>> ENCHANTMENT_COMPONENT_TYPES = DeferredRegister.create(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, "examplemod");
+
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<Unit>> EXAMPLE =
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<Unit>builder()
+            .persistent(Unit.CODEC)
+            .build());
+```
+
+### Testing For Enchantment Components
+To check if an enchantment component is present on a given item stack, one must first get the item's `minecraft:enchantment` component. The value of this component is an `ItemEnchantments` instance containing a set of all the Enchantments on the item, which can be acquired using `ItemEnchantments#keySet()`. The enchantment components of a given enchantment can then be acquired using the `Enchantment::effects()` method:
```

Comment:
This should be done with either `runIterationOnItem`, `runIterationOnEquipement`, `pickHighestLevel`, `has`, or `hasTag`. These more or less handle what you are trying to write, which would generally be more limiting as most enchantments are supposed to return some value based on the iterable run.

Reply:
I... did not know `EnchantmentHelper#has` existed.